### PR TITLE
Feature/role user

### PR DESCRIPTION
**Pull request by Vitalii02**
Here it was done creating roles and checking for what a person has access to a resource

изображение

### CODE REVIEW

File: backend/src/common/guards/role.guard.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        context: ExecutionContext,
+    ): Promise<boolean> {
+        const isAdmin = this.reflector.get<boolean>('admin', context.getHandler());
+        if (!isAdmin) {
+            return true;
+        }
+        const request = context.switchToHttp().getRequest();
+        const userId = request.user.id;
+        const userRole = await this.roleRepository.findOne(userId);
+        return userRole && userRole.position === 'admin';
```
Comment (kstroevsky) [inline backend/src/common/guards/role.guard.ts:23]:
```suggestion
        if (!this.reflector.get<boolean>('admin', context.getHandler())) {
            return true;
        }
        
        const userId = context.switchToHttp().getRequest().user.id;
        const userRole = await this.roleRepository.findOne(userId);
        return userRole && userRole.position === 'admin';
```

File: backend/src/main.ts [hunk 2]
```diff
@@ -21,6 +21,6 @@ async function bootstrap() {
       whitelist: true,
     }),
   )
-  await app.listen(port);
+  await app.listen(3000);
```
Comment (kstroevsky) [inline backend/src/main.ts:24]: PORT to environment variables to control it from one-point-access.

**Review by kstroevsky — CHANGES_REQUESTED** [review]
LGTM

File: backend/src/main.ts (hunk 2 above)
Comment (Vitalii02) [reply backend/src/main.ts:24]: fixed

**Pull request merged**